### ISSUE TITLE
Failed to receive server version information when trying the IB historic download example

### ISSUE THREAD

**Issue by ashkan-leo** [nautechsystems/nautilus_trader#2047]
Hi folks,

I am a noob, and I am trying to run the historic_download.py example to gather some historical data from Interactive Broker.
When reaching `await client.connect()` I get the following errors:

```
2024-11-08T07:35:58.073533880Z [INFO] TRADER-000.MessageBus: config.database=None
2024-11-08T07:35:58.073559910Z [INFO] TRADER-000.MessageBus: config.encoding='msgpack'
2024-11-08T07:35:58.073562745Z [INFO] TRADER-000.MessageBus: config.timestamps_as_iso8601=False
2024-11-08T07:35:58.073567064Z [INFO] TRADER-000.MessageBus: config.buffer_interval_ms=None
2024-11-08T07:35:58.073569588Z [INFO] TRADER-000.MessageBus: config.autotrim_mins=None
2024-11-08T07:35:58.073572073Z [INFO] TRADER-000.MessageBus: config.use_trader_prefix=True
2024-11-08T07:35:58.073574548Z [INFO] TRADER-000.MessageBus: config.use_trader_id=True
2024-11-08T07:35:58.073577393Z [INFO] TRADER-000.MessageBus: config.use_instance_id=False
2024-11-08T07:35:58.073581511Z [INFO] TRADER-000.MessageBus: config.streams_prefix='stream'
2024-11-08T07:35:58.073585418Z [INFO] TRADER-000.MessageBus: config.types_filter=None
2024-11-08T07:35:58.073596429Z [INFO] TRADER-000.Cache: READY
2024-11-08T07:35:58.073615916Z [INFO] TRADER-000.InteractiveBrokersClient-005: READY
2024-11-08T07:35:58.074645704Z [INFO] TRADER-000.InteractiveBrokersClient-005: RUNNING
2024-11-08T07:35:58.075087283Z [INFO] TRADER-000.InteractiveBrokersClient-005: Starting InteractiveBrokersClient (5)...
2024-11-08T07:35:58.075109906Z [INFO] TRADER-000.InteractiveBrokersClient-005: Connecting to 127.0.0.1:4002 with client id: 5
2024-11-08T07:35:58.078160513Z [WARN] TRADER-000.InteractiveBrokersClient-005: Failed to receive server version information. Retries remaining: 4.
2024-11-08T07:35:59.079322942Z [WARN] TRADER-000.InteractiveBrokersClient-005: Failed to receive server version information. Retries remaining: 3.
2024-11-08T07:36:00.081946644Z [WARN] TRADER-000.InteractiveBrokersClient-005: Failed to receive server version information. Retries remaining: 2.
2024-11-08T07:36:01.084183387Z [WARN] TRADER-000.InteractiveBrokersClient-005: Failed to receive server version information. Retries remaining: 1.
2024-11-08T07:36:02.086370181Z [WARN] TRADER-000.InteractiveBrokersClient-005: Failed to receive server version information. Retries remaining: 0.
2024-11-08T07:36:03.087788119Z [ERROR] TRADER-000.InteractiveBrokersClient-005: Connection failed: Max retry attempts reached. Failed to receive server version information.
2024-11-08T07:36:03.088922424Z [INFO] TRADER-000.InteractiveBrokersClient-005: Resetting InteractiveBrokersClient (5)...
2024-11-08T07:36:03.088936772Z [INFO] TRADER-000.InteractiveBrokersClient-005: Stopping InteractiveBrokersClient (5)...
2024-11-08T07:36:03.089286647Z [ERROR] TRADER-000.InteractiveBrokersClient-005: Error occurred while canceling tasks: An asyncio.Future, a coroutine or an awaitable is required
TypeError(An asyncio.Future, a coroutine or an awaitable is required)
Traceback (most recent call last):
  File "/home/ashkan/w/trade/.venv/lib/python3.12/site-packages/nautilus_trader/adapters/interactive_brokers/client/client.py", line 266, in _stop_async
    await asyncio.gather(*tasks, return_exceptions=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/asyncio/tasks.py", line 831, in gather
    fut = ensure_future(arg, loop=loop)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/asyncio/tasks.py", line 689, in ensure_future
    raise TypeError('An asyncio.Future, a coroutine or an awaitable '
```

I appreciate any advice on how to make this work :)

### Specifications

- OS platform: NixOS
- Python version: 3.12.6
- `nautilus_trader` version: 1.205.0

**Comment by cjdsellers** [comment]
Hi @ashkan-leo 

Thanks for reaching out.

@rsmb7z or @benjaminsingleton any ideas on this one? 🙏

**Comment by rsmb7z** [comment]
Hi @ashkan-leo
Are you using dockered gateway? Can you test running the gateway or TWS first making sure its successfully logged in after authentication and then run ` historic_download.py`.

**Comment by ashkan-leo** [comment]
Thanks @rsmb7z 
I am using the dockered gateway. 
I am running on a headless server. I can't run TWS on it. TWS works fine on my other systems though. 
My guess is there is a problem with authentication as well. I didn't get a second factor on my phone.

**Comment by rsmb7z** [comment]
Hi @ashkan-leo 
You can try increasing `DockerizedIBGatewayConfig.timeout` value.

**Comment by ashkan-leo** [comment]
Can you advise on how to confirm authentication is successful? 

Asking because 2FA is not received. do we think timeout is the issue we are not being authenticated correctly?

**Comment by rsmb7z** [comment]
To continue troubleshooting, you'll need to connect via VNC to investigate the gateway. Please [follow the instructions](https://github.com/gnzsnz/ib-gateway-docker/pkgs/container/ib-gateway) to set up the container and VNC connection.

**Comment by ashkan-leo** [comment]
Hi @rsmb7z,
Thanks for your suggestion. I wanted to share my progress so far:
1. I setup a docker compose file -- similar to what is in the [instructions here](https://github.com/gnzsnz/ib-gateway-docker/pkgs/container/ib-gateway)
2. I also set the trading mode to live. When setting to paper I see that I don't get the 2FA
3. Now, when the docker container is running we connect to port = 4001 -- instead of firing a new DockerizedIBGateway in the script.
4. At this point things mostly work. I am able to execute most of the things in the script.

What I am unable to execute though are the following lines:
```python
    bars = await client.request_bars(
        bar_specifications=["1-HOUR-LAST", "30-MINUTE-MID"],
        start_date_time=datetime.datetime(2023, 11, 6, 9, 30),
        end_date_time=datetime.datetime(2023, 11, 6, 16, 30),
        tz_name="America/New_York",
        contracts=[contract],
        instrument_ids=[instrument_id],
    )

    trade_ticks = await client.request_ticks(
        "TRADES",
        start_date_time=datetime.datetime(2023, 11, 6, 10, 0),
        end_date_time=datetime.datetime(2023, 11, 6, 10, 1),
        tz_name="America/New_York",
        contracts=[contract],
        instrument_ids=[instrument_id],
 ```

With the error ` Failed to request historical ticks:No market data permissions for ISLAND STK `. I believe this is due to me not enabling market data permissions. Would be great to get some confirmation from the experts here.

My goal at this point is to get some stuff running to assess the feasibility of trading -- as such I'm not too eager to pay to get data or anything like that -- one reason I ended up here as opposed to say QuantConnect, QuantRocket, etc. I'm wondering what would be the next step for me, and I'd appreciate any advise.
